awswrangler/s3/_read_deltalake.py

```diff
@@ -0,0 +1,124 @@
+"""Amazon S3 Read Delta Lake Module (PRIVATE)."""
+
+import logging
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import pandas as pd
+import pyarrow.fs as pa_fs
+from deltalake import DataCatalog, DeltaTable
+
+_logger: logging.Logger = logging.getLogger(__name__)
+
+
+def read_deltalake(
+    path: str,
+    version: Optional[int] = None,
+    storage_options: Optional[Dict[str, str]] = None,
+    without_files: bool = False,
+    partitions: Optional[List[Tuple[str, str, Any]]] = None,
+    columns: Optional[List[str]] = None,
+    filesystem: Optional[Union[str, pa_fs.FileSystem]] = None,
```

Comment:
`awswrangler` is entirely based on boto3 for credentials management. Potentially we could build the Pyarrow FileSystem from the boto3 session. However, it might not be necessary altogether (see comment on `to_pandas` below).

Reply:
I changed the implementation by relying only on the boto3 session to build the `storage_options` for the storage back of the `DeltaTable` while keeping the possibility to add more configuration via `s3_additional_kwargs`.